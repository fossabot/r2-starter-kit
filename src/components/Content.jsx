import React, { Component } from 'react';
import PropTypes from 'prop-types';

import FadeOutImage from 'components/FadeOutImage';

export default class Content extends Component {
    render() {
        const {
            id,
            title,
            author = {
                name: '',
                email: ''
            },
            body
        } = this.props;
        return (
            <div>
                {id &&
                    <div className="ts card text container">
                        <div className="image">
                            <FadeOutImage
                              url={`https://picsum.photos/1200/720?image=${id}`}
                              placeholderUrl={`https://picsum.photos/60/36?image=${id}`}
                            />
                        </div>
                        <div className="content">
                            <div className="header">
                                {title}
                            </div>
                            <div className="bulleted meta">
                                {`${author.name} (${author.email})`}
                            </div>
                            <div className="description" dangerouslySetInnerHTML={{ __html: body }} />
                        </div>
                    </div>
                }
            </div>
        );
    }
    static propTypes = {
        id: PropTypes.number,
        title: PropTypes.string,
        author: PropTypes.shape({
            name: PropTypes.string,
            email: PropTypes.string
        }),
        body: PropTypes.string
    }
}
