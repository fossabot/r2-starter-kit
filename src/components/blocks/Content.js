import React, { PureComponent } from 'react';
import PropTypes from 'prop-types';

import FadeOutImage from 'components/elements/FadeOutImage';

export default class Content extends PureComponent {
  static propTypes = {
    authorName: PropTypes.string,
    authorEmail: PropTypes.string,
    title: PropTypes.string,
    description: PropTypes.string,
    image: PropTypes.string,
    placeholder: PropTypes.string,
  };
  static defaultProps = {
    authorName: '',
    authorEmail: '',
    title: '',
    description: '',
    image: '',
    placeholder: '',
  };
  render() {
    const {
      authorName,
      authorEmail,
      title,
      description,
      image,
      placeholder,
    } = this.props;
    return (
      <div>
        <div className="ts card text container">
          <div className="image">
            <FadeOutImage
              url={image}
              placeholder={placeholder}
            />
          </div>
          <div className="content">
            <div className="header">{title}</div>
            <div className="bulleted meta">
              {`${authorName} (${authorEmail})`}
            </div>
            <div className="description">
              {description}
            </div>
          </div>
        </div>
      </div>
    );
  }
}
