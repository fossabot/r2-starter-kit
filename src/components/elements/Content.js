import React, { PureComponent } from 'react';
import PropTypes from 'prop-types';

import FadeOutImage from 'components/elements/FadeOutImage';

export default class Content extends PureComponent {
  static propTypes = {
    id: PropTypes.number,
    title: PropTypes.string,
    author: PropTypes.shape({
      name: PropTypes.string,
      email: PropTypes.string,
    }),
    body: PropTypes.string,
  };
  static defaultProps = {
    id: null,
    title: '',
    author: {
      name: '',
      email: '',
    },
    body: '',
  };
  render() {
    const {
      id,
      title,
      author,
      body,
    } = this.props;
    return (
      <div>
        {id && (
          <div className="ts card text container">
            <div className="image">
              <FadeOutImage
                url={`https://picsum.photos/1200/720?image=${id}`}
                placeholder={`https://picsum.photos/60/36?image=${id}`}
              />
            </div>
            <div className="content">
              <div className="header">{title}</div>
              <div className="bulleted meta">
                {`${author.name || ''} (${author.email || ''})`}
              </div>
              <div
                className="description"
                dangerouslySetInnerHTML={{ __html: body }}
              />
            </div>
          </div>
        )}
      </div>
    );
  }
}
