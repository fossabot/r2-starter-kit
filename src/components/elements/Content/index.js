import React, { PureComponent } from 'react';
import PropTypes from 'prop-types';

import ProgressiveImage from 'components/elements/ProgressiveImage';

export default class Content extends PureComponent {
  static propTypes = {
    authorName:  PropTypes.string,
    authorEmail: PropTypes.string,
    title:       PropTypes.string,
    description: PropTypes.string,
    image:       PropTypes.string,
    placeholder: PropTypes.string,
  };
  static defaultProps = {
    authorName:  '',
    authorEmail: '',
    title:       '',
    description: '',
    image:       '',
    placeholder: '',
  };
  render() {
    const {
      authorName,
      authorEmail,
      title,
      description,
      image,
      placeholder,
    } = this.props;
    return (
      <div>
        <div className="ts card text container">
          <div className="image">
            <ProgressiveImage
              url={image}
              placeholder={placeholder}
            />
          </div>
          <div className="content">
            <div className="header">{title}</div>
            <div className="bulleted meta">
              {`${authorName} (${authorEmail})`}
            </div>
            <div className="description">
              {description}
            </div>
          </div>
        </div>
      </div>
    );
  }
}
