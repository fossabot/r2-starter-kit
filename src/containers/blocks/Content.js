import { connect } from 'react-redux';

import Content from 'components/blocks/Content';
import { getLargeImage, getTinyImage } from 'utils/normalizer';

function mapStateToProps(state) {
  return {
    authorName: state.response.user.data.name,
    authorEmail: state.response.user.data.email,
    title: state.response.post.data.title,
    description: state.response.post.data.body,
    image: getLargeImage(state.response.post.data.id),
    placeholder: getTinyImage(state.response.post.data.id),
  };
}

export default connect(mapStateToProps)(Content);
