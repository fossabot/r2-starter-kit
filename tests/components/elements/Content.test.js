import React from 'react';
import { shallow } from 'enzyme';
import Content from 'components/elements/Content';

describe('<Content />', () => {
  const initProps = {
    id: 1,
    title: '',
    author: {},
    body: '',
  };

  let wrapper;
  beforeAll(() => {
    wrapper = shallow(<Content {...initProps} />);
  });

  it('should render initial component', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should render correct component with assigned values', () => {
    wrapper.setProps({
      id: 2,
      title: 'Post Title',
      author: {
        name: 'Tony Pai',
        email: 'tonypai@me.com',
      },
      body: 'Post Content',
    });
    expect(wrapper).toMatchSnapshot();
  });
});
