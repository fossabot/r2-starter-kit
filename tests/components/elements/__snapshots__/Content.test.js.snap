// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Content /> should render correct component with assigned values 1`] = `
<div>
  <div
    className="ts card text container"
  >
    <div
      className="image"
    >
      <FadeOutImage
        placeholder="https://picsum.photos/60/36?image=2"
        url="https://picsum.photos/1200/720?image=2"
      />
    </div>
    <div
      className="content"
    >
      <div
        className="header"
      >
        Post Title
      </div>
      <div
        className="bulleted meta"
      >
        Tony Pai (tonypai@me.com)
      </div>
      <div
        className="description"
        dangerouslySetInnerHTML={
          Object {
            "__html": "Post Content",
          }
        }
      />
    </div>
  </div>
</div>
`;

exports[`<Content /> should render initial component 1`] = `
<div>
  <div
    className="ts card text container"
  >
    <div
      className="image"
    >
      <FadeOutImage
        placeholder="https://picsum.photos/60/36?image=1"
        url="https://picsum.photos/1200/720?image=1"
      />
    </div>
    <div
      className="content"
    >
      <div
        className="header"
      />
      <div
        className="bulleted meta"
      >
         ()
      </div>
      <div
        className="description"
        dangerouslySetInnerHTML={
          Object {
            "__html": "",
          }
        }
      />
    </div>
  </div>
</div>
`;
